src/main/java/com/mcmiddleearth/mcmescripts/compiler/EntityCompiler.java

```diff
@@ -33,27 +35,30 @@ public class EntityCompiler {
     private static final int DEFAULT_SPAWN_DISTANCE   = 64;
 
     public static Set<Trigger> compile(JsonObject jsonData) {
+        Collection<Condition> conditions = ConditionCompiler.compile(jsonData);
```

Comment:
Scripts can already have a `condition` and `conditions` property on them, which will cause the script to load/unload:

https://github.com/CarperCreative/MCME-Scripts/blob/180d4cd1e380b2480138624f775553544802b6b7/src/main/java/com/mcmiddleearth/mcmescripts/script/Script.java#L47